src/StackExchange.Redis/ResultProcessor.cs

```diff
@@ -1743,6 +1749,59 @@ protected override bool SetResultCore(PhysicalConnection connection, Message mes
             }
         }
 
+        internal sealed class StreamAutoClaimProcessor : StreamProcessorBase<StreamAutoClaimResult>
+        {
+            private readonly bool processIdsOnly;
+
+            public StreamAutoClaimProcessor(bool processIdsOnly = false)
+            {
+                this.processIdsOnly = processIdsOnly;
+            }
+
+            protected override bool SetResultCore(PhysicalConnection connection, Message message, in RawResult result)
+            {
+                // See https://redis.io/commands/xautoclaim for command documentation.
+
+                if (!result.IsNull && result.Type == ResultType.MultiBulk)
+                {
+                    var items = result.GetItems();
+
+                    // [0] The next start ID.
+                    var nextStartId = items[0].AsRedisValue();
+
+                    // [1] The array of either StreamEntry's or message IDs.
+                    var entriesOrIds = items[1].GetItems();
+
+                    // [2] The array of message IDs deleted from the stream that were in the PEL.
+                    //     This is not available in 6.2 so we need to be defensive when reading
+                    //     this part of the response.
+                    RedisValue[] deletedIds = null!;
```

Comment:
@NickCraver I see what you're saying, but since this property is only set in >= 7.0, the compiler was complaining that it could be null. Hence the null forgiveness and the guard when populating the final result.

Reply:
I'm not sure how that would happen, maybe a state on the way here? It should be `RedisValue[]?` and the `?? Array.Empty<RedisValue>()` will handle the case feeding into the result, never a null usage.

Reply:
@NickCraver This is probably more in line with what you want, right? No more null forgiveness operators.

```csharp
RedisValue[]? deletedIds = null;
// ...snip...
if (items.Length == 3)
{
    deletedIds = items[2].GetItemsAsValues();
}
SetResult(message, new StreamAutoClaimResult(
  nextStartId, 
  arr, 
  deletedIds ?? Array.Empty<RedisValue>()));
```

Reply:
Yep!